drivers/bbram/bbram_xec.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021 Microchip Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_bbram
+
+#include <drivers/bbram.h>
+#include <errno.h>
+#include <soc.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
+
+/** Device config */
+struct bbram_xec_config {
+	/** BBRAM base address */
+	uintptr_t base;
+	/** BBRAM size (Unit:bytes) */
+	int size;
+};
+
+static int bbram_xec_check_invalid(const struct device *dev)
+{
+	struct vbatr_regs *const regs = (struct vbatr_regs *)(DT_REG_ADDR_BY_NAME(
+					DT_INST(0, microchip_xec_pcr), vbatr));
+
+	if (regs->PFRS & BIT(MCHP_VBATR_PFRS_VBAT_RST_POS)) {
+		regs->PFRS |= BIT(MCHP_VBATR_PFRS_VBAT_RST_POS);
+		return -EFAULT; /* VBAT power rail failure */
+	}
+
+	return 0;
+}
+
+static int bbram_xec_get_size(const struct device *dev, size_t *size)
+{
+	const struct bbram_xec_config *dcfg =
+		(const struct bbram_xec_config *)dev->config;
+
+	*size = dcfg->size;
+	return 0;
+}
+
+static int bbram_xec_read(const struct device *dev, size_t offset, size_t size,
+			  uint8_t *data)
+{
+	const struct bbram_xec_config *dcfg =
+		(const struct bbram_xec_config *)dev->config;
+
+	if (size < 1 || offset + size > dcfg->size) {
+		return -EFAULT;
+	}
+
+	bytecpy(data, ((uint8_t *)dcfg->base + offset), size);
```

Comment:
why not define `base` using the right type? (seems `uint8_t *`).

Reply:
defined base as `uint8_t *`